packages/utils/src/lib/reports/formatting.unit.test.ts

```diff
@@ -126,100 +128,161 @@ describe('formatSourceLine', () => {
+  it('should not format the file path as a link when outputDir is undefined (VS Code)', () => {
+    vi.stubEnv('TERM_PROGRAM', 'vscode');
+    vi.stubEnv('GITHUB_ACTIONS', 'false');
+
     expect(
       linkToLocalSourceForIde({
-        file: '/packages/utils/src/index.ts',
+        file: toUnixPath('/packages/utils/src/index.ts'),
       }).toString(),
     ).toBe('`/packages/utils/src/index.ts`');
   });
 
-  it('should format file path as link when outputDir is provided', () => {
+  it('should format the file path as a link when outputDir is defined (VS Code)', () => {
+    vi.stubEnv('TERM_PROGRAM', 'vscode');
+    vi.stubEnv('GITHUB_ACTIONS', 'false');
+    const filePath = toUnixPath('/packages/utils/src/index.ts');
+    const outputDir = toUnixPath('/.code-pushup');
+
     expect(
-      linkToLocalSourceForIde(
-        { file: '/packages/utils/src/index.ts' },
-        { outputDir: '/.code-pushup' },
-      ).toString(),
+      linkToLocalSourceForIde({ file: filePath }, { outputDir }).toString(),
     ).toBe('[`/packages/utils/src/index.ts`](../packages/utils/src/index.ts)');
   });
 
-  it('should return link to specific line when startLine is provided (VS Code)', () => {
+  it('should return a link to a specific line when startLine is provided (VS Code)', () => {
     vi.stubEnv('TERM_PROGRAM', 'vscode');
+    vi.stubEnv('GITHUB_ACTIONS', 'false');
+    const filePath = toUnixPath('/packages/utils/src/index.ts');
+    const outputDir = toUnixPath('/.code-pushup');
+
     expect(
       linkToLocalSourceForIde(
-        {
-          file: '/packages/utils/src/index.ts',
-          position: { startLine: 2 },
-        },
-        { outputDir: '/.code-pushup' },
+        { file: filePath, position: { startLine: 2 } },
+        { outputDir },
       ).toString(),
     ).toBe(
       '[`/packages/utils/src/index.ts`](../packages/utils/src/index.ts#L2)',
     );
-    vi.unstubAllEnvs();
   });
 });
 
-describe('formatFilePosition', () => {
+describe('getGitHubLink', () => {
+  beforeEach(() => {
+    vi.stubEnv('TERM_PROGRAM', '');
+    vi.stubEnv('GITHUB_ACTIONS', 'true');
+    vi.stubEnv('GITHUB_SERVER_URL', 'https://github.com');
+    vi.stubEnv('GITHUB_REPOSITORY', 'user/repo');
+    vi.stubEnv('GITHUB_SHA', '1234567890abcdef');
+  });
+
   it.each([
-    ['../src/index.ts', { startLine: 2 }, '../src/index.ts#L2'],
-    ['../src/index.ts', { startLine: 2, endLine: 5 }, '../src/index.ts#L2-L5'],
-    ['../src/index.ts', { startLine: 2, startColumn: 1 }, '../src/index.ts#L2'],
+    [
+      { startLine: 2 },
+      'https://github.com/user/repo/blob/1234567890abcdef/.code-pushup/src/index.ts#L2',
+    ],
+    [
+      { startLine: 2, endLine: 5 },
+      'https://github.com/user/repo/blob/1234567890abcdef/.code-pushup/src/index.ts#L2-L5',
+    ],
+    [
+      { startLine: 2, startColumn: 1 },
+      'https://github.com/user/repo/blob/1234567890abcdef/.code-pushup/src/index.ts#L2',
```

Comment:
I actually just found out that GitHub URLs support columns as well as line numbers :open_mouth: E.g.: https://github.com/code-pushup/cli/blob/main/packages/utils/src/lib/merge-configs.ts#L38C60-L40C10

Do you think you could include the columns, too? :pray:

Reply:
Done ✅